### PR TITLE
Fix login failure 

### PR DESCRIPTION
**Pull request by eisuke**
With a user who email address already exists, but different uid.

@cookpad/dev-infra 👓

### CODE REVIEW

**Comment by takai** [comment]
It's obviously another account. I do not prefer this changes because it can cause unintended consequence.
You may want to create tiny script to deal with this kind of thing.

**Comment by eisuke** [comment]
> It's obviously another account.

Yes, I know it as ideal. The reason for taking this approach is
* This case occurred more than I think. (ie. part-time employee to regular employee)
* I trust that accounts are under control in G Suite hosted domain.

But I agree this approach is not good. So I redoed it ae49794 (Removing table constaints)
What do you think? @takai @cookpad/dev-infra

**Comment by takai** [comment]
It doesn't look bad 👍

File: db/migrate/027_drop_uniq_constraint_user_email.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+class DropUniqConstraintUserEmail < ActiveRecord::Migration
+  def change
```
Comment (eagletmt) [inline db/migrate/027_drop_uniq_constraint_user_email.rb]: This should be `def up` because this migration is irreversible.

Comment (eisuke) [inline db/migrate/027_drop_uniq_constraint_user_email.rb]: nice catch! fixed e803981

File: app/models/kuroko2/user.rb [hunk 2]
```diff
@@ -17,8 +17,8 @@ class Kuroko2::User < Kuroko2::ApplicationRecord
   has_many :admin_assignments, dependent: :restrict_with_error
   has_many :assigned_job_definitions, through: :admin_assignments, source: :job_definition
 
-  validates :name, uniqueness: { case_sensitive: false} , presence: true
-  validates :email, uniqueness: { case_sensitive: false}, presence: true
+  validates :name,  uniqueness: { case_sensitive: false }, presence: true, unless: :google_account?
```
Comment (eagletmt) [inline app/models/kuroko2/user.rb]: The presence should be validated even if `google_account?`

Comment (eisuke) [inline app/models/kuroko2/user.rb]: 🙈 149ea35

**Pull request merged**